### PR TITLE
rack adjustments

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Update rack content rendering
- Fix racks showing up in scan tool
-

[ ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java [hunk 1]
```diff
@@ -595,27 +612,60 @@ private void refreshTextureCache()
...
+                        else
+                        {
+                            itemQueue.add(block);
+                        }
+                    }
+                }
+
+                if (!isBlockItem)
+                {
+                    if (alreadyAddedItem)
```
Comment (Mekle001) [inline src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java:661]: Did you miss a "!" here? This logic feels very backwards, and hard to follow.

Comment (Raycoms) [reply src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java:661]: Not, itentional. If we already added it, we stop here and just add it to the queue, else we pass and mark true

**Review by someaddons — APPROVED** [review]
